build/builder.go

```diff
@@ -266,8 +267,8 @@ func checkCTags() string {
 
 // SetDefaults sets reasonable default options.
 func (o *Options) SetDefaults() {
-	if o.CTags == "" {
-		o.CTags = checkCTags()
+	if o.CTagsPath == "" && !o.DisableCTags {
+		o.CTagsPath = checkCTags()
```

Comment:
If we only do this change, does it fix the problem? Having to check two fields everywhere is not ideal. IE if DisableCtags is true force CTagsPath to be empty in SetDefaults?

Reply:
You need to keep this change in SetDefaults and getting rid of the bit below in the NewBuilder(): constructor in order to make the newly added tests pass.

```
if b.opts.DisableCTags {
		b.opts.CTags = ""
	}
```

However, there are other usages of CTags path that are essentially checking to see whether or not ctags is enabled at all. In order to properly validate that you need to see both 1) whether or not we explicitly disabled ctags 2) whether or not we found a valid ctags binary. So, I think it's worth leaving both checks in here for all such places for correctness' sake. 

However, I do agree with you that it's awkward / error prone to need to check two fields. As a follow up, I can try experimenting with replacing both `ctagsPath` and `disableCtags` as an interface that captures both semantics.